Better error handling when opening/writing config files, fixes  #1177

andrewfrench opened the pull request:
## The Problem/Issue/Bug:
#1177: Naive error checking caused nil pointer dereference panics when the error returned was indicating something other than the file not existing.
 
## How this PR Solves The Problem:
This change prevents the panic by expanding the error case to _any_ error value, not just those indicating the file doesn't exist. In the specific case encountered in #1177, this means that if the file exists, but ddev doesn't have permissions to read or edit it, we won't try to.

## Manual Testing Instructions:
- `mkdir -p panic/sites/default`
- `cd panic`
- `sudo chown root sites`
- `sudo chmod 000 sites`
- `ddev config --project-name panic --docroot . --project-type drupal8`
- Ensure that ddev warns that it doesn't have permissions to write config files and .gitignore, but no panic is encountererd

## Automated Testing Overview:
<!-- Please provide an overview of tests introduced by this PR, or an explanation for why no tests are needed. -->

## Related Issue Link(s):
#1177

rfay commented on pkg/ddevapp/drupal.go:
```diff
@@ -653,8 +653,7 @@ func drupalEnsureWritePerms(app *DdevApp) error {
-		// If the file doesn't exist, don't try to set the permissions.
-		if os.IsNotExist(err) {
+		if err != nil {
 			continue
```
I'm pretty sure this should emit a util.Warning*(), since we're skipping by here and never hitting anything that will give them a warning about what's happened.

andrewfrench commented:
I've addressed the comments in here that have to do with better logging and warning, and have improved a few warning messages that were already returning a PathError and don't need to restate the path to the file in question.

I think refactoring this is a good idea and I support it, but I'm not really open to doing that as a hop-on for this PR - I think it's best to let this solve the original issue and open another to capture what needs to be done to consolidate logic moving forward.

rfay commented:
Oops, I messed up. With this we get warnings about files that don't exist, we'll need to not emit warnings for a file that doesn't exist.
```
Unable to ensure write permissions: stat /Users/rfay/workspace/nblewis.com/sites/default/settings.ddev.php: no such file or directory
Ensuring write permissions for nblewis.com
Unable to ensure write permissions: stat /Users/rfay/workspace/nblewis.com/sites/default/settings.php: no such file or directory
Unable to ensure write permissions: stat /Users/rfay/workspace/nblewis.com/sites/default/settings.ddev.php: no such file or directory
Unable to ensure write permissions: stat /Users/rfay/workspace/nblewis.com/sites/default/services.yml: no such file or directory
No settings.php file exists, creating one
```
